packages/amplify_authenticator/lib/src/blocs/auth/auth_bloc.dart

```diff
@@ -32,14 +32,15 @@ class StateMachineBloc {
   StreamSink<AuthEvent> get authEvent => _authEventController.sink;
 
   /// Outputs events into the event transformer.
-  late final Stream<AuthEvent> _authEventStream = _authEventController.stream;
+  Stream<AuthEvent> get authEventStream => _authEventController.stream;
```

Comment:
*Note:* This was made public so that the Verify User View Model can listen to it and update state when certain events are emitted. This is a new pattern. The authenticator currently doesn't have a need to pass state to view models.

The state that is being passed is the list of unverified attributes. See `AuthSetUnverifiedAttributeKeys` farther down in this file as well as the `VerifyUserViewModel` when it is listened to.

Reply:
I would just create a new AuthUnverifiedUser state in place of AuthFlow.verifyUser and add the keys as data to that state object. This starts to get a little unwieldy since bloc are supposed to be a black box which translate events to states.

Reply:
That is how I implemented it initially actually. I would not be opposed to going back to that. I [have a branch](https://github.com/aws-amplify/amplify-flutter/compare/amplify-authenticator...Jordan-Nelson:feat/authenticator/verify-user) up with that approach (that branch is definitely in a less polished draft state that this branch fyi). 

I switched it because I felt like that approach was a little confusing with the amount of layers that the data had to be passed through (VerifyUserState -> build method of the AmplifyAuthenticator -> VerifiedUserScreen -> VerifiedUserForm -> set state on VerifyUserViewModel) but I could simplify that by having the VerifiedUserScreen update the VerifyUserViewModel with the initial data.

I probably won't get back to this PR for a couple days, but I might go with that approach when I get back to it.